src/k2/cmd/tpcc/tpcc_client.cpp

```diff
@@ -246,15 +265,15 @@ class Client {
                 uint32_t w_id = (seastar::this_shard_id() % _max_warehouses()) + 1;
                 uint32_t d_id = (seastar::this_shard_id() % _districts_per_warehouse()) + 1;
                 TPCCTxn* curTxn;
-                if (txn_type <= 43) {
+                if (txn_type <= _weights[0]) {
```

Comment:
I don't think this is equivalent to the previous code. If your first txn is of largest weight (say 51), then you will never run any other transactions.

You should add up the weights to 100 and then pick a txn type based on where the random number falls (as the previous code did)

Reply:
the logic should be the same since the _weights actually holds the partial sum for each iteration in the parsing logic instead of the weight itself.